container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_aggregator_functions/implementations/vector/vector_logs_aggregator_container.go

```diff
@@ -0,0 +1,90 @@
+package vector
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_manager"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/object_attributes_provider"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+
+type vectorLogsAggregatorContainer struct{}
+
+func NewVectorLogsAggregatorContainer() *vectorLogsAggregatorContainer {
+	return &vectorLogsAggregatorContainer{}
+}
+
+func (vectorContainer *vectorLogsAggregatorContainer) CreateAndStart(
+	ctx context.Context,
+	portNumber uint16,
+	targetNetworkId string,
+	objAttrsProvider object_attributes_provider.DockerObjectAttributesProvider,
+	dockerManager *docker_manager.DockerManager,
+) (
+	resultContainerId string,
+	resultContainerLabels map[string]string,
+	resultRemoveLogsAggregatorContainerFunc func(),
+	resultErr error,
+) {
+	vectorContainerConfigProviderObj := createVectorContainerConfigProvider(portNumber)
+
+	logsAggregatorVolumeAttrs, err := objAttrsProvider.ForLogsAggregatorVolume()
+	if err != nil {
+		return "", nil, nil, stacktrace.Propagate(err, "An error occurred getting the logs collector volume attributes")
+	}
+	volumeName := logsAggregatorVolumeAttrs.GetName().GetString()
+	volumeLabelStrs := map[string]string{}
+	for labelKey, labelValue := range logsAggregatorVolumeAttrs.GetLabels() {
+		volumeLabelStrs[labelKey.GetString()] = labelValue.GetString()
+	}
+	if err := dockerManager.CreateVolume(ctx, volumeName, volumeLabelStrs); err != nil {
```

Comment:
Is it possible that the volume already exists? Since the volume is at the engine level, what happens if we delete the engine? Do we also delete the volume?

Reply:
I removed the volume code in this PR - going to add it back in the v1 where I add persistent logs storage via persistent volume